### PR TITLE
Condition middleware

### PR DESCRIPTION
**Pull request by KeenS**
Add a new middleware that let you conditionally enables another middleware
See #934 .

### CODE REVIEW

**Comment by JohnTitor** [comment]
Could you also update CHANGES.md?

**Comment by KeenS** [comment]
Updated. I added a new title "not yet released". Is this correct?

**Review by fafhrd91 — APPROVED** [review]

**Comment by fafhrd91** [comment]
lgtm!

File: src/middleware/condition.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
+use actix_service::{Service, Transform};
```
Comment (JohnTitor) [inline src/middleware/condition.rs:2]:
```suggestion
//! `Middleware` for conditionally enables another middleware.
use actix_service::{Service, Transform};
```

File: src/middleware/condition.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+use futures::future::{ok, Either, FutureResult, Map};
+use futures::{Future, Poll};
+
+/// `Middleware` for conditianally enables another middleware
```
Comment (JohnTitor) [inline src/middleware/condition.rs]:
```suggestion
/// `Middleware` for conditionally enables another middleware.
```

File: src/middleware/condition.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+/// use actix_web::middleware::{Condition, NormalizePath};
+/// use actix_web::App;
+///
+/// fn main() {
+///     let enable_normalize = std::env::var("NORMALIZE_PATH") == Ok("true".into());
+///     let app = App::new()
+///         .wrap(Condition::new(enable_normalize, NormalizePath));
+/// }
+/// ```
+///
```
Comment (JohnTitor) [inline src/middleware/condition.rs]: This line is unnecessary.
```suggestion
```

File: src/middleware/condition.rs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+use futures::future::{ok, Either, FutureResult, Map};
+use futures::{Future, Poll};
+
+/// `Middleware` for conditianally enables another middleware
+/// The controled middleware must not change the `Service` interfaces.
+/// This means you cannot control such middlewares like `Logger` or `Compress`.
+///
+/// ## Usage
+///
+/// ```
```
Comment (JohnTitor) [inline src/middleware/condition.rs]: IMO: We have explicit `rust` annotation in other places so it seems reasonable to add it here.
```suggestion
/// ```rust
```

**Review by JohnTitor — COMMENTED** [review]
LGTM with some nits

**Comment by KeenS** [comment]
All suggestions has been applied.

**Review by JohnTitor — APPROVED** [review]
Thanks!

**Comment by fafhrd91** [comment]
thanks!

**Pull request merged**